src/ai/ai.cpp

```diff
@@ -1692,27 +1714,24 @@ dcon::cb_type_id pick_fabrication_type(sys::state& state, dcon::nation_id from,
 	}
 }
 
-bool valid_construction_target(sys::state& state, dcon::nation_id from, dcon::nation_id target) {
+static inline bool valid_construction_target(sys::state& state, dcon::nation_id from, dcon::nation_id target) {
+	// Copied from commands.cpp:can_fabricate_cb()
 	if(from == target)
 		return false;
-
+	if(state.world.nation_get_constructing_cb_type(from))
+		return false;
 	auto ol = state.world.nation_get_overlord_as_subject(from);
 	if(state.world.overlord_get_ruler(ol) && state.world.overlord_get_ruler(ol) != target)
 		return false;
-	auto sl = state.world.nation_get_in_sphere_of(target);
-	if(sl == from)
-		return false;
-	if(nations::are_allied(state, target, from))
+	if(state.world.nation_get_in_sphere_of(target) == from)
 		return false;
-	if(nations::are_allied(state, sl, from))
+	if(military::are_at_war(state, target, from))
 		return false;
-	if(state.world.nation_get_military_score(target) > state.world.nation_get_military_score(from))
+	
+	if(estimate_defensive_strength(state, target) * 0.5f > estimate_defensive_strength(state, from))
```

Comment:
Comparing my defensive strength to their defensive strength is not a good measure for when I might be able to use a cb -- I can't call my defensive allies in to attack necessarily